nucypher-core-python/src/lib.rs

```diff
@@ -633,6 +633,128 @@ impl ReencryptionResponse {
     }
 }
 
+//
+// Request Keys
+//
+
+#[pyclass(module = "nucypher_core")]
+#[derive(derive_more::From, derive_more::AsRef)]
+pub struct RequestSharedSecret {
+    backend: nucypher_core::RequestSharedSecret,
+}
+
+#[pyclass(module = "nucypher_core")]
+#[derive(Clone, PartialEq, Eq, derive_more::From, derive_more::AsRef)]
+pub struct RequestPublicKey {
+    backend: nucypher_core::RequestPublicKey,
+}
+
+#[pymethods]
+impl RequestPublicKey {
+    #[staticmethod]
+    pub fn from_bytes(data: &[u8]) -> PyResult<Self> {
+        from_bytes::<_, nucypher_core::RequestPublicKey>(data)
+    }
+
+    fn __bytes__(&self) -> PyObject {
+        to_bytes(self)
+    }
+
+    fn __richcmp__(&self, other: &Self, op: CompareOp) -> PyResult<bool> {
+        richcmp(self, other, op)
+    }
+
+    fn __hash__(&self) -> PyResult<isize> {
+        hash("RequestPublicKey", self)
+    }
+
+    fn __str__(&self) -> PyResult<String> {
+        Ok(format!("{}", self.backend))
+    }
+}
+
+#[pyclass(module = "nucypher_core")]
+#[derive(derive_more::From, derive_more::AsRef)]
+pub struct RequestSecretKey {
+    backend: nucypher_core::RequestSecretKey,
+}
+
+#[pymethods]
+impl RequestSecretKey {
+    #[staticmethod]
+    pub fn random() -> PyResult<Self> {
+        Ok(Self {
+            backend: nucypher_core::RequestSecretKey::random(),
+        })
+    }
+
+    pub fn public_key(&self) -> RequestPublicKey {
+        RequestPublicKey {
+            backend: self.backend.public_key(),
+        }
+    }
+
+    pub fn diffie_hellman(&self, their_public_key: &RequestPublicKey) -> RequestSharedSecret {
+        RequestSharedSecret {
+            backend: self.backend.diffie_hellman(their_public_key.as_ref()),
+        }
+    }
+
+    fn __str__(&self) -> PyResult<String> {
+        Ok(format!("{}", self.backend))
+    }
+}
+
+#[pyclass(module = "nucypher_core")]
+#[derive(derive_more::From, derive_more::AsRef)]
+pub struct RequestKeyFactory {
+    backend: nucypher_core::RequestKeyFactory,
+}
+
+#[pymethods]
+impl RequestKeyFactory {
+    #[staticmethod]
+    pub fn random() -> PyResult<Self> {
+        Ok(Self {
+            backend: nucypher_core::RequestKeyFactory::random(),
+        })
+    }
+
+    #[staticmethod]
+    pub fn seed_size() -> usize {
+        nucypher_core::RequestKeyFactory::seed_size()
+    }
+
+    #[staticmethod]
+    pub fn from_secure_randomness(seed: &[u8]) -> PyResult<Self> {
+        let factory = nucypher_core::RequestKeyFactory::from_secure_randomness(seed)
+            .map_err(|err| PyValueError::new_err(format!("{}", err)))?;
+        Ok(Self { backend: factory })
+    }
```

Comment:
We could shave-off a few characters here and there by introducing this helper function:

```suggestion
   // Elsewhere
   fn map_py_value_err<T: fmt::Display>(err: T) -> PyErr {
     PyValueError::new_err(format!("{err}"))
   }
   
   // Here
    #[staticmethod]
    pub fn from_secure_randomness(seed: &[u8]) -> PyResult<Self> {
        let backend = nucypher_core::RequestKeyFactory::from_secure_randomness(seed)
            .map_err(map_py_value_err)?;
        Ok(Self { backend })
    }
```